packages/salesforcedx-vscode-core/test/vscode-integration/orgBrowser/metadataCmp.test.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { expect } from 'chai';
+import * as fs from 'fs';
+import * as path from 'path';
+import { SinonStub, stub } from 'sinon';
+import { isNullOrUndefined } from 'util';
+import {
+  buildComponentsList,
+  ForceListMetadataExecutor,
+  getComponentsPath
+} from '../../../src/orgBrowser';
+import { getRootWorkspacePath, OrgAuthInfo } from '../../../src/util';
+
+describe('Force List Metadata', () => {
+  it('Should build list metadata command', async () => {
+    const outputPath = 'outputPath';
+    const metadataType = 'ApexClass';
+    const defaultUsername = 'test-username1@example.com';
+    const forceListMetadataExec = new ForceListMetadataExecutor(
+      metadataType,
+      outputPath,
+      defaultUsername
+    );
+    const forceDescribeMetadataCmd = forceListMetadataExec.build({});
+    expect(forceDescribeMetadataCmd.toCommand()).to.equal(
+      `sfdx force:mdapi:listmetadata -m ${metadataType} -u ${defaultUsername} -f ${outputPath} --json --loglevel fatal`
+    );
+  });
+
+  it('Should build list metadata command with folder arg', async () => {
+    const outputPath = 'outputPath';
+    const metadataType = 'Report';
```

Comment:
We should test all of the "folder types" to make sure none of them have a command built without a folder param. Perhaps in the command file there could be an exported set of those types and iterate through each one expecting the right command. Not to mention when building the command (line 55, metadataCmp.ts), the set can be used to check for membership of a given type instead of having a conditional check for each type built.

Reply:
Yeah, I think this is a good idea!